pepsico/monthly_seasonal_conv_mapping.py

```diff
@@ -0,0 +1,199 @@
+
+    # Assign the season column with correct length
+    seasonal_avgs_df['season'] = ['DJF', 'JFM', 'FMA', 'MAM', 'AMJ', 'MJJ',
+                                   'JJA', 'JAS', 'ASO', 'SON', 'OND', 'NDJ']
+
+    return seasonal_avgs_df
+
+
+def write_xarray_to_csv(ds, scenario, model, variable, output_dir):
+    df = ds.to_dataframe().reset_index()
+    file_path = f'{output_dir}/{scenario}_{model}_{variable}_annual_monthly_avg.csv'
+    df.to_csv(file_path, index=False)
+
+def write_dataframe_to_csv(df, scenario, model, variable, output_dir):
+    file_path = f'{output_dir}/{scenario}_{model}_{variable}_seasonal_avg.csv'
+    df.to_csv(file_path, index=False)
+
+def map_averages(ds, variable):
+    
+    #different than compute_annual_monthly_avg because keeps the spatial component when averaging
+    #this will be inputted into the plotting function
+    
+    # Convert daily data to monthly
+    monthly_ds = ds.resample(T="1M").mean()
+
+    # Compute the average for each month across all years
+    monthly_avg = monthly_ds.groupby("T.month").mean(dim="T")
+
+    return monthly_avg
+
+
+def plot_monthly(ds, variable, scenario, model, output_dir, month):
+    #convert units if needed
+    units = ds[variable].attrs.get('units', 'unknown')
+    
+    monthly_data = ds.sel(month=month)
+    
+    if variable == 'pr':
+        fig = px.imshow(
+            monthly_data[variable],
+            labels={'color': f'{variable} ({units})'},
+            title=f'Monthly Average ({variable})  - Month {month}',
+            origin='lower',
+            zmin=0,
+            zmax=15
+        )
+    else:
+        fig = px.imshow(
+            monthly_data[variable],
+            labels={'color': f'{variable} ({units})'},
+            title=f'Monthly Average ({variable})  - Month {month}',
+            origin='lower'
+        )
+    fig.show() 
+    output_file = f"{output_dir}/{variable}_{scenario}_{model}_monthly_avg_map_{month}.html"
+    fig.write_html(file=output_file)
+    print("Printed")
+
+
+def plot_seasonal(ds, variable, scenario, model, output_dir, season):
+    units = ds[variable].attrs.get('units', 'unknown')
+    # Determine season indices based on the selected season
+    season_indices = {
+        'DJF': [11, 0, 1],  # Dec, Jan, Feb
+        'JFM': [0, 1, 2],   # Jan, Feb, Mar
+        'FMA': [1, 2, 3],   # Feb, Mar, Apr
+        'MAM': [2, 3, 4],   # Mar, Apr, May
+        'AMJ': [3, 4, 5],   # Apr, May, Jun
+        'MJJ': [4, 5, 6],   # May, Jun, Jul
+        'JJA': [5, 6, 7],   # Jun, Jul, Aug
+        'JAS': [6, 7, 8],   # Jul, Aug, Sep
+        'ASO': [7, 8, 9],   # Aug, Sep, Oct
+        'SON': [8, 9, 10],  # Sep, Oct, Nov
+        'OND': [9, 10, 11], # Oct, Nov, Dec
+        'NDJ': [10, 11, 0]  # Nov, Dec, Jan
+    }
+    # Select data for the specific season
+    seasonal_data = ds.sel(T=ds['T.month'].isin(season_indices[season]))
+    seasonal_avg = seasonal_data.mean(dim='T')
```

Comment:
nice, but you had done all the work with your rolling function already. Only you are not using it because you stuck the spatial average in it. If you were taking the spatial average out of that, then you could use the same function to compute seasonal stuff.